src/pages/workspace/content/allContents/index.tsx

```diff
@@ -0,0 +1,211 @@
+];
+
+const AllContentsPage = () => {
+  const [selectedKey, setSelectedKey] = useState("allContents");
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [filter, setFilter] = useState("all");
+  const [sortBy, setSortBy] = useState("updated");
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const handleSearch = (search: string) => {
+    setSearchTerm(search.toLowerCase());
+  };
+
+  const handleFilterChange = (filter: string) => {
+    setFilter(filter);
+  };
+
+  const handleSortChange = (sortBy: string) => {
+    setSortBy(sortBy);
+  };
+
+  const filteredData = useMemo(
+    () =>
+      sampleData.filter((content) =>
+        content.title.toLowerCase().includes(searchTerm)
+      ),
+    [searchTerm]
+  );
+
+  const displayedRows = filteredData.slice(
+    page * rowsPerPage,
+    page * rowsPerPage + rowsPerPage
+  );
+
+  useEffect(() => {
+    const getContentList = async () => {
+      try {
+        const reqBody = {
+          request: {
+            filters: {
+              status: [
+                "Draft",
+                "FlagDraft",
+                "Review",
+                "Processing",
+                "Live",
+                "Unlisted",
+                "FlagReview",
+              ],
+              createdBy: "84721b4a-6536-4cb0-b8c3-57583ef4cada",
+              primaryCategory: [
+                "Course Assessment",
+                "eTextbook",
+                "Explanation Content",
+                "Learning Resource",
+                "Practice Question Set",
+                "Teacher Resource",
+                "Exam Question",
+                "Content Playlist",
+                "Course",
+                "Digital Textbook",
+                "Question paper",
+              ],
+            },
+            offset: 0,
+            limit: 9,
+            query: "",
+            sort_by: {
+              lastUpdatedOn: "desc",
+            },
+          },
+        };
+        const response = await getContent(reqBody);
+        console.log(response);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    getContentList();
+  }, []);
+
+  return (
+    <Layout selectedKey={selectedKey} onSelect={setSelectedKey}>
+      <Box p={3}>
+        <Typography variant="h4">Show all Content</Typography>
+        <Typography mb={2}>Here you see all your content.</Typography>
+
+        <Box mb={3}>
+          <SearchBox
+            placeholder="Search by title..."
+            onSearch={handleSearch}
+            onFilterChange={handleFilterChange}
+            onSortChange={handleSortChange}
+          />
+        </Box>
+
+        <Table>
+          <TableHead>
```

Comment:
@suvarnakale for table use same library used in admin portal

Reply:
ok

Reply:
use KA table library but our table has complex data and icons, KA is not supporting it